src/osd/OSDMap.cc

```diff
@@ -1369,7 +1369,7 @@ void OSDMap::set_max_osd(int m)
   osd_state.resize(max_osd);
   osd_weight.resize(max_osd);
   for (; o<max_osd; o++) {
-    osd_state[o] = 0;
+    osd_state[o] = 0; // clear state bits e.g. CEPH_OSD_EXISTS
```

Comment:
@tchaikov 
My initial target is to comment the meaning of setting ```osd_state[index]=0```.
"0" doesn't hint the purpose.
What's about below one?
```c
osd_state.resize(max_osd, 0); // clear the extended state bits e.g. CEPH_OSD_EXISTS
osd_weight.resize(max_osd, CEPH_OSD_OUT); 
```

Reply:
> @tchaikov
> My initial target is to comment the meaning of setting `osd_state[index]=0`.
> "0" doesn't hint the purpose.
> What's about below one?

i knew. i wanted to go further.

> 
> ```c++
> osd_state.resize(max_osd, 0); // clear the extended state bits e.g. CEPH_OSD_EXISTS
> osd_weight.resize(max_osd, CEPH_OSD_OUT); 
> ```

i don't think "clear" is the right word here. what we are doing here is to initialize the states of the newly created osd  with an **empty** state instead of clearing them. but i like the way to be more explicit.

Reply:
@changchengx i agree with you that
```c++
osd_state.resize(max_osd, 0);
osd_weight.resize(max_osd, CEPH_OSD_OUT); 
```
in better. could you incorporate this change into your changeset?

Reply:
@tchaikov Done